homeassistant/components/media_player/kodi.py

```diff
@@ -262,6 +296,17 @@ def on_hass_stop(event):
         else:
             self._ws_server = None
 
+        # Script creation for the turn on/off config options
+        if turn_on_action is not None:
+            turn_on_action = script.Script(
+                self.hass, turn_on_action,
+                "Kodi Turn ON action script",
+                partial(self.async_update_ha_state, force_refresh=True))
```

Comment:
Since `async_update_ha_state` is a coroutine, you shouldn't need to use partial here. You can just pass it as `self.async_update_ha_state(True)`. It won't get executed until `Script` schedules the job.